mayastor/src/rebuild.rs

```diff
@@ -93,23 +125,36 @@ pub struct RebuildJob {
     segment_size_blks: u64,
     tasks: RebuildTasks,
     complete_fn: fn(String, String) -> (),
+    /// channel used to signal rebuild completion
     pub complete_chan: (Sender<RebuildState>, Receiver<RebuildState>),
+    /// current state of the rebuild job
     pub state: RebuildState,
 }
 
+/// Place holder for rebuild statistics
 pub struct RebuildStats {}
 
+/// Public facing operations on a Rebuild Job
 pub trait RebuildOperations {
+    /// Collects statistics from the job
     fn stats(&self) -> Option<RebuildStats>;
+    /// Schedules the job to start in a future and returns a complete channel
+    /// which can be waited on
     fn start(&mut self) -> Receiver<RebuildState>;
+    /// Stops the job which then triggers the completion hooks
     fn stop(&mut self) -> Result<(), RebuildError>;
+    /// Temporarily pauses the job until resume is called
```

Comment:
I would remove "Temporarily" here. It is permanently paused until another operation moves it out of this state. Maybe we should also remove the reference to resume, because stop could also be issued while paused.

Reply:
Done